src/main/java/seedu/commands/EditCommand.java

```diff
@@ -37,59 +45,74 @@ public EditCommand(String fullCommand) {
     public void execute(IncomeList incomes, SpendingList spendings) {
         assert incomes != null;
         assert spendings != null;
-        String[] userInputWords = fullCommand.split(" ", 5);
         try {
-            if (userInputWords.length < 5) {
-                throw new WiagiMissingParamsException(INCORRECT_PARAMS_NUMBER
-                        + EDIT_COMMAND_FORMAT);
-            }
-            if (userInputWords[1].equals("spending")) {
-                editList(userInputWords, spendings);
-            } else if (userInputWords[1].equals("income")) {
-                editList(userInputWords, incomes);
-            } else {
-                throw new WiagiInvalidInputException(INVALID_CATEGORY + EDIT_COMMAND_FORMAT);
-            }
+            commandHandler(incomes, spendings);
         } catch (WiagiMissingParamsException | WiagiInvalidInputException | WiagiInvalidIndexException e) {
             Ui.printWithTab(e.getMessage());
         }
     }
 
+    private void commandHandler(IncomeList incomes, SpendingList spendings)
```

Comment:
Do we want to considered making a class for commandHandler and extractArguments similar to Command

Reply:
can add as a new issue if needed

Reply:
Thought of doing that, but it may increase complexity, as `commandHandler` have different method signatures across different commands. 

Furthermore, some of the commands does not require these methods, as the implementation is very simple, such as `ListAllCommand` and `ByeCommand`.